logging/troubleshooting/cluster-logging-troubleshooting-for-critical-alerts.adoc

```diff
@@ -0,0 +1,495 @@
+
+.. Delete the index.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name> -X DELETE
+----
+
+ . After freeing up the disk space, when used disk percentage drops below 90%, unblock write to this particular node.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_all/_settings?pretty -X PUT -d '{"index.blocks.read_only_allow_delete": null}'
+----
+
+[id="elasticsearch-jvm-heap-use-is-high"]
+== Elasticsearch JVM Heap Use is High
+
+The Elasticsearch node JVM Heap memory used is above 75%.
+
+=== Troubleshooting
+
+Consider https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#heap-size-settings[increasing the heap size].
+
+[id="aggregated-logging-system-cpu-is-high"]
+== Aggregated Logging System CPU is High
+
+System CPU usage on the node is high.
+
+=== Troubleshooting
+
+Check the CPU of the cluster node. Consider allocating more CPU resources to the node.
+
+[id="elasticsearch-process-cpu-is-high"]
+== Elasticsearch Process CPU is High
+
+Elasticsearch process CPU usage on the node is high.
+
+=== Troubleshooting
+
+Check the CPU of the cluster node. Consider allocating more CPU resources to the node.
+
+[id="elasticsearch-disk-space-is-running-low"]
+== Elasticsearch Disk Space is Running Low
+
+The Elasticsearch Cluster is predicted to be out of disk space within the next 6 hours.
+
+The prediction is made based on the current trend in occupying the disk space on that node.
+
+=== Troubleshooting
+
+. Get the disk space of the Elasticsearch node.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_nodes/stats/fs?pretty
+----
+
+. In the command output, check the `nodes.node_name.fs` field to determine the free disk space on that node.
+
+. Increase the disk space on all nodes.
+
+. If increasing the disk space is not possible, add a new data node to the cluster.
+
+. If adding a new data node is problematic, decrease the total cluster redundancy policy.
+
+.. Check the current `redundancyPolicy`.
++
+[source,terminal]
+----
+oc edit es elasticsearch -n openshift-logging
+----
++
+NOTE: If using Cluster Logging Custom Resource, then:
++
+[source,terminal]
+----
+oc edit cl instance -n openshift-logging
+----
+
+.. If cluster redundancy policy is higher than `SingleRedundancy`, set it to `SingleRedundancy` and save it.
+
+. If nothing above helps, delete old indices.
+
+.. Check the status of all indices on Elasticsearch.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- indices
+----
+
+.. Identify the index that can be deleted.
+
+.. Delete the index.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name> -X DELETE
+----
```

Comment:
Hi @sasagarw, will the changes in https://issues.redhat.com/browse/LOG-1090 be merged into 5.1? Currently, the changes are not in `release-5.1` branch. 
If no, I think we need to document down the query string to release locks on indices.  cc @rolfedh 
```
es_util --query=*/_settings?pretty -X PUT -d '{"index.blocks.read_only_allow_delete": null}'
```